cmd/kubeadm/app/phases/addons/proxy/proxy.go

```diff
@@ -47,16 +47,16 @@ const (
 )
 
 // EnsureProxyAddon creates the kube-proxy addons
-func EnsureProxyAddon(cfg *kubeadmapi.ClusterConfiguration, localEndpoint *kubeadmapi.APIEndpoint, client clientset.Interface) error {
+func EnsureProxyAddon(cfg *kubeadmapi.ClusterConfiguration, localEndpoint *kubeadmapi.APIEndpoint, client clientset.Interface, printManifest bool) error {
```

Comment:
instead of using `fmt.Printf/ln` we should use pass a io.Writer to EnsureProxyAddon and EnsureDNSAddon functions.
and use 

kubeadm commands pass io.Writer from the command construction:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/kubeadm.go#L49